api/nodes/utils.py

```diff
@@ -56,3 +58,25 @@ def get_file_object(target, path, provider, request):
         return waterbutler_request.json()['data']
     except KeyError:
         raise ServiceUnavailableError(detail='Could not retrieve files information at this time.')
+
+
+class NodeOptimizationMixin(object):
+    """Mixin with convenience method for optimizing serialization of nodes.
+    Annotates the node queryset with several properties to reduce number of queries.
+    """
+    def optimize_node_queryset(self, queryset):
+        auth = get_user_auth(self.request)
+        admin_scope = has_admin_scope(self.request)
+        abstract_node_contenttype_id = ContentType.objects.get_for_model(AbstractNode).id
+        guid = Guid.objects.filter(content_type_id=abstract_node_contenttype_id, object_id=OuterRef('parent_id'))
+        parent = NodeRelation.objects.annotate(parent__id=Subquery(guid.values('_id')[:1])).filter(child=OuterRef('pk'), is_node_link=False)
+        wiki_addon = WikiNodeSettings.objects.filter(owner=OuterRef('pk'), deleted=False)
+        contribs = Contributor.objects.filter(user=auth.user, node=OuterRef('pk'))
+        return queryset.prefetch_related('root').prefetch_related('subjects').annotate(
```

Comment:
Nit: I believe you can pass any number of kwargs to `.annotate`, so you don't need to chain it a bunch of times. Doesn't really matter though.